src/agent/testresultreader.ts

```diff
@@ -0,0 +1,404 @@
+                var errorMessage = "";
+                if(testCaseNode.failure) {
+                    outcome = "Failed";
+                    errorMessage = testCaseNode.failure.text();
+                }
+                else if(testCaseNode.error) {
+                    outcome = "Failed";
+                    errorMessage = testCaseNode.error.text();
+                }
+
+                var testResult : ifm.TestRunResult = <ifm.TestRunResult> {
+                    state: "Completed",
+                    computerName: hostName,
+                    resolutionState: null,
+                    testCasePriority: 1,
+                    failureType: null,
+                    automatedTestName: testName,
+                    automatedTestStorage: testStorage,
+                    automatedTestType: "JUnit",
+                    automatedTestTypeId: null,
+                    automatedTestId: null,
+                    area: null,
+                    owner: buildRequestedFor, 
+                    runBy: buildRequestedFor,
+                    testCaseTitle: testName,
+                    revision: 0,
+                    dataRowCount: 0,
+                    testCaseRevision: 0,
+                    outcome: outcome,
+                    errorMessage: errorMessage,
+                    durationInMs: testCaseDuration * 1000, //convert to milliseconds
+                };
+                
+                testResults.push(testResult);
+            }
+        }
+
+        if(totalRunDuration < totalTestCaseDuration) {
+            totalRunDuration = totalTestCaseDuration; //run duration may not be set in the xml, so use teh testcase duration
+        }
+
+        var completedDate = timeStamp;
+        completedDate.setSeconds(timeStamp.getSeconds() + totalRunDuration);
+
+        //create test run data
+        var testRun: ifm.TestRun = <ifm.TestRun>    {
+            name: runName,
+            iteration: "",
+            state: "InProgress",
+            automated: true,
+            errorMessage: "",
+            type: "",
+            controller: "",
+            buildDropLocation: "",
+            buildPlatform: platform,
+            buildFlavor: config,
+            comment: "",
+            testEnvironmentId: "",
+            startDate: timeStamp,
+            //completeDate: completedDate,
+            releaseUri: "",
+            build: { id: buildId }
+        };
+
+        testRun2 = <ifm.TestRun2>{
+            testRun: testRun,
+            testResults: testResults,
+        };         
+
+        return testRun2;
+    }      
+}
+
+//-----------------------------------------------------
+// Read NUnit results from a file
+// - file: string () - location of the NUnit results file 
+//-----------------------------------------------------
+export class NUnitResultReader implements trp.IResultReader {
+    
+    public readResults(file: string, runContext: trp.TestRunContext) : Q.Promise<ifm.TestRun2> {
```

Comment:
repeated code, I am a fan of DRY, probably could use some refactoring?  Maybe extract those two functions into a base class?